build_bins.sh

```diff
@@ -1,354 +1,90 @@
-fi
-FILES="jce_policy-8.zip $FILES"
-
-# Pull all the (unversioned) gems required for the cluster 
-for i in patron wmi-lite simple-graphite ruby-augeas chef-rewind; do
-  if ! [[ -f gems/${i}.gem ]]; then
-    gem fetch ${i}
-    ln -s ${i}-*.gem ${i}.gem || true
-  fi
-  FILES="${i}*.gem $FILES"
-done
-
-# Get the Rubygem for mysql2
-if ! [[ -f gems/mysql2.gem ]]; then
-  gem fetch mysql2 -v 0.4.4
-  ln -s mysql2-*.gem mysql2.gem || true
-fi
-FILES="mysql2*.gem $FILES"
-
-# Get the Rubygem for kerberos
-if ! [[ -f gems/rake-compiler.gem ]]; then
-  gem fetch rake-compiler
-  ln -s rake-compiler*.gem rake-compiler.gem || true
-fi
-FILES="rake-compiler*.gem $FILES"
-
-# Get the Rubygem for sequel
-if ! [[ -f gems/sequel.gem ]]; then
-  gem fetch sequel -v 4.36.0
-  ln -s sequel-*.gem sequel.gem || true
-fi
-FILES="sequel*.gem $FILES"
-
-# Get the Rubygem for rkerberos
-if ! [[ -f gems/rkerberos.gem ]]; then
-  gem fetch rkerberos
-  ln -s rkerberos*.gem rkerberos.gem || true
-fi
-FILES="rkerberos*.gem $FILES"
-
-# Get the Rubygem for webhdfs
-if ! [[ -f gems/webhdfs.gem ]]; then
-  gem fetch webhdfs -v 0.5.5
-  ln -s webhdfs-*.gem webhdfs.gem || true
-fi
-FILES="webhdfs*.gem $FILES"
-
-# Get Rubygem for zabbixapi
-if ! [[ -f gems/zabbixapi.gem ]]; then
-  gem fetch zabbixapi -v ${ZABBIXAPI_VERSION}
-  ln -s zabbix*.gem zabbixapi.gem || true
-fi
-FILES="zabbix*.gem $FILES"
-
-# Get the Rubygem for zookeeper
-if ! [[ -f gems/zookeeper.gem ]]; then
-  gem fetch zookeeper -v 1.4.7
-  ln -s zookeeper-*.gem zookeeper.gem || true
-fi
-FILES="zookeeper*.gem $FILES"
-
-# Fetch the cirros image for testing
-if ! [[ -f cirros-0.3.0-x86_64-disk.img ]]; then
-  while ! $(file cirros-0.3.0-x86_64-disk.img | grep -q 'QEMU QCOW Image'); do
-    $CURL -O -L https://launchpad.net/cirros/trunk/0.3.0/+download/cirros-0.3.0-x86_64-disk.img
-  done
-fi
-FILES="cirros-0.3.0-x86_64-disk.img $FILES"
-
-# Grab the Ubuntu 14.04 installer image with a 4.4 "HWE" kernel from Xenial
-TRUSTY_IMAGE="ubuntu-14.04-hwe44-mini.iso"
-if ! [[ -f $TRUSTY_IMAGE ]]; then
-    while ! $(file $TRUSTY_IMAGE | grep -qE '(x86 boot sector)|(ISO 9660 CD-ROM)'); do
-	$CURL -o $TRUSTY_IMAGE http://archive.ubuntu.com/ubuntu/dists/trusty-updates/main/installer-amd64/current/images/xenial-netboot/mini.iso
-    done
-fi
-FILES="$TRUSTY_IMAGE $FILES"
-
-# Grab the Ubuntu 12.04 installer image with a 3.13 kernel.
-PRECISE_IMAGE="ubuntu-12.04-hwe313-mini.iso"
-if ! [[ -f $PRECISE_IMAGE ]]; then
-    while ! $(file $PRECISE_IMAGE | grep -qE '(x86 boot sector)|(ISO 9660 CD-ROM)'); do
-	$CURL -o $PRECISE_IMAGE http://archive.ubuntu.com/ubuntu/dists/precise-updates/main/installer-amd64/current/images/trusty-netboot/mini.iso
-    done
-fi
-FILES="$PRECISE_IMAGE $FILES"
+pushd $DIR/bins/ > /dev/null
 
+if [ $(dpkg-query -W -f='${Status}' chefdk 2>/dev/null | grep -c 'ok installed') -eq 0 ]; then
+    wget -nc https://opscode-omnibus-packages.s3.amazonaws.com/ubuntu/14.04/x86_64/chefdk_0.12.0-1_amd64.deb
```

Comment:
It frustrates me that $CURL is set up to disable SSL verification, but I suppose it doesn't matter, since we checksum the package with sha256sum.